lib/fizzy/execute.cpp

```diff
@@ -611,7 +611,7 @@ execution_result execute(Instance& instance, FuncIdx func_idx, span<const uint64
     const auto& code = instance.module.codesec[code_idx];
     auto* const memory = instance.memory.get();
 
-    std::vector<uint64_t> locals(args.size() + code.local_count, 0);
+    std::vector<Value> locals(args.size() + code.local_count);
```

Comment:
Doesn't this leave them uninitialized, because `Value` default constructor doesn't initialize it?

Reply:
No.
1. Vector is using _value initialization_ and then _copy_ to init elements.
2. For unions, value initialization is _zero initialization_, i.e.
3. > If T is a union type, the first non-static named data member is zero-initialized and all padding is initialized to zero bits.

Reply:
It looks the spec for it has changed in C++11, but the result is the same. But having/not-having default constructor that initializes with zero affects GCC code generation (looks like a bug rather) and the version without zeroing default constructor is faster as it fills locals by a loop instead of a `memset` call.